src/main/java/org/elasticsearch/index/query/CommonTermsQueryParser.java

```diff
@@ -117,9 +98,9 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                     } else if ("high_freq_operator".equals(currentFieldName) || "highFreqOperator".equals(currentFieldName)) {
                         String op = parser.text();
                         if ("or".equalsIgnoreCase(op)) {
-                            highFreqOccur = BooleanClause.Occur.SHOULD;
+                            highFreqOperator = Operator.OR;
                         } else if ("and".equalsIgnoreCase(op)) {
-                            highFreqOccur = BooleanClause.Occur.MUST;
+                            highFreqOperator = Operator.AND;
```

Comment:
maybe we could add a fromString(String) method to the Operator enum instead, so we don't have to repeat the same code twice